fix: Wallet Attestation response code and method (#287)
italia/eidas-it-wallet-docs: docs/en/wallet-attestation.rst

**Step 9**: The Wallet Instance sends the ``challenge`` with Key Attestation and Cryptographic Hardware Key Tag to the Wallet Provider Backend in order to register the Wallet Instance identified with the Cryptographic Hardware Key public key.

> The request to the Wallet Provider endpoint in order to register the Wallet Instance MUST use the HTTP POST method with the parameters in the message body encoded in ``application/json`` format and the following parameters MUST be provided:

Reviewer: ```suggestion
In order to register the Wallet Instance, the request to the Wallet Provider MUST use the HTTP POST method. The parameters MUST be encoded using the `application/json` format and included in the message body. The following parameters MUST be provided:
```

I'd propose to give a name to the wallet registration endpoint in the wallet solution architecture

.. _table_http_request_claim:
.. list-table:: Wallet Instance registration http request parameters
    :widths: 20 60 20
    :header-rows: 1

    * - **Claim**
      - **Description**
      - **Reference**
    * - **challenge**
      - MUST be set to the challenge obtained from the Wallet Provider throught the ``nonce`` endpoint.
      - `OAuth 2.0 Nonce Endpoint`_
    * - **key_attestation**
      - It MUST be a ``base64url`` encoded Key Attestation obtained from the **Device Integrity Service**.
      -
    * - **hardware_key_tag**

>       - It MUST be set to the unique identifier of the **Cryptographic Hardware Keys** in ``base64url`` encoded

Reviewer: ```suggestion
      - It MUST be set with the unique identifier of the **Cryptographic Hardware Keys** and encoded in ``base64url``.
```

  4. If these checks are passed, it MUST register the Wallet Instance, keeping the Cryptographic Hardware Key Tag and all useful information related to the device.
  5. It SHOULD associate the Wallet Instance with a specific User uniquely identified within the Wallet Provider's systems. This will be useful for the lifecycle of the Wallet Instance and for a future revocation.

> The response, in case of success, from the Wallet Provider must necessarily be a 204 No content.

Reviewer: ```suggestion
Upon successful registration of the Wallet Instance, the Wallet Provider MUST respond with a status code set to 204 (No Content).
```

  }

> **Step 18**: The Wallet Instance receives the Wallet Attestation signed by the Wallet Provider and performs security and integrity verifications.
> The response is returned by the Wallet Provider at the end of the issuance flow.
> If the authentication is successful the http response code MUST be 200 OK.

Reviewer: ```suggestion
**Step 18**: The response is returned by the Wallet Provider. If successful, the HTTP response code MUST be set to 200 OK and contain the Wallet Attestation signed by the Wallet Provider. The Wallet Instance therefore performs security and integrity verification about the Wallet Attestation. 
```

here we should be more accurate providing which security and integrity verification are supposed to be made by the wallet instance about the wallet attestation received

Below is a non-normative example of the response.

.. code-block:: http

    HTTP/1.1 200 OK
    Content-Type: application/jwt

>     eyJhbGciOiJFUzI1NiIsInR5cCI6IndhbGx... redacted

Reviewer: ```suggestion
    eyJhbGciOiJFUzI1NiIsInR5cCI6IndhbGx ... 
```
